test/modules/generic-mqtt-tester/src/message_initiator.rs

```diff
@@ -54,10 +59,20 @@ impl MessageInitiator {
         let payload_size = self.settings.message_size_in_bytes() as usize;
         let dummy_data = &vec![b'a'; payload_size];
         loop {
-            info!("publishing message {}", seq_num);
+            if let Some(messages_to_send) = self.settings.messages_to_send() {
```

Comment:
I believe you could use power of deref in Option :). If the setting is `None` it will never be true.
```rust
    if Some(seq_num) = self.settings.messages_to_send() {
        ...
        break;
    }
```